service/history/workflow/mutable_state_impl.go

```diff
@@ -598,23 +598,40 @@ func (ms *MutableStateImpl) ChasmTree() historyi.ChasmTree {
 // GetNexusCompletion converts a workflow completion event into a [nexus.OperationCompletion].
 // Completions may be sent to arbitrary third parties, we intentionally do not include any termination reasons, and
 // expose only failure messages.
-func (ms *MutableStateImpl) GetNexusCompletion(ctx context.Context) (nexus.OperationCompletion, error) {
+func (ms *MutableStateImpl) GetNexusCompletion(
+	ctx context.Context,
+	requestID string,
+) (nexus.OperationCompletion, error) {
 	ce, err := ms.GetCompletionEvent(ctx)
 	if err != nil {
 		return nil, err
 	}
+
 	// Create the link information about the workflow to be attached to fabricated started event if completion is
 	// received before start response.
-	startLink := temporalnexus.ConvertLinkWorkflowEventToNexusLink(&commonpb.Link_WorkflowEvent{
+	link := &commonpb.Link_WorkflowEvent{
 		Namespace:  ms.namespaceEntry.Name().String(),
 		WorkflowId: ms.executionInfo.WorkflowId,
 		RunId:      ms.executionState.RunId,
+		// Backwards compatibility: this is the default link type.
 		Reference: &commonpb.Link_WorkflowEvent_EventRef{
 			EventRef: &commonpb.Link_WorkflowEvent_EventReference{
 				EventType: enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED,
 			},
 		},
-	})
+	}
+	requestIDInfo := ms.executionState.RequestIds[requestID]
```

Comment:
I don't think this works across CaN and workflow retries.

Reply:
It doesn't work, if the workflow continued-as-new, retried, or reseted, it will generate a link to the new workflow.

Reply:
Discussed offline, this is okay.